solution

ViktoriiaPitsan opened the pull request:
[DEMO LINK](https://ViktoriiaPitsan.github.io/react_list-of-goods-hooks/)

Pitsan-Viktoriia committed:
solution

Moroz-Dmytro commented on src/App.tsx:
```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
+  None = 'none',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+  Reverse = 'reverse',
```
```suggestion
```

You don't use it

Moroz-Dmytro commented on src/App.tsx:
```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState<boolean>(false);
```
```suggestion
  const [isReversed, setIsReversed] = useState<boolean>(false);
```

Moroz-Dmytro commented on src/App.tsx:
```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
+  const [reversed, setReversed] = useState<boolean>(false);
+  const [initialGoods] = useState(goodsFromServer);
+
+  const visibleGoods = getPreparedGoods(initialGoods, { sortField, reversed });
```
```suggestion
  const visibleGoods = getPreparedGoods(goodsFromServer, { sortField, reversed });
```

Why do you need this?

Moroz-Dmytro commented on src/App.tsx:
```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
+
+  const isDifferent = !initialGoods.every(
+    (good, i) => good === visibleGoods[i],
+  );
```
```suggestion
  const isResetButtonVisible = sortField !== SortType.None || isReversed
```

Moroz-Dmytro reviewed (changes requested):
Good work 👍 
But fix some comments below

ViktoriiaPitsan committed:
Update src/App.tsx

Co-authored-by: Dmytro Moroz <86789161+Moroz-Dmytro@users.noreply.github.com>

Pitsan-Viktoriia committed:
Refactor sorting and reset logic:
- Renamed reversed to isReversed
- Removed unnecessary initialGoods state
- Simplified reset button visibility check

Pitsan-Viktoriia committed:
Merge branch 'develop' of https://github.com/ViktoriiaPitsan/react_list-of-goods-hooks into develop

ViktoriiaPitsan commented:
@Moroz-Dmytro Thank you for your feedback! I’ve made the changes you suggested.

781calmstate commented on src/App.tsx:
```diff
@@ -15,36 +16,112 @@ export const goodsFromServer = [
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
```
not a mistake, but typescript is smart enough to understand that your state gonna be boolean if you pass some boolean value as default value

781calmstate commented on src/App.tsx:
```diff
@@ -15,36 +16,112 @@ export const goodsFromServer = [
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
```
good job on moving this functionality out of component, it looks very clean and easy to read

781calmstate reviewed (approved):
Good job! Great solution 👍
